miden-tx/src/host/account_delta.rs

```diff
@@ -125,6 +125,49 @@ impl<A: AdviceProvider> TransactionHost<A> {
         Ok(())
     }
 
+    /// Extracts information from the process state about the storage map being updated and
+    /// records the latest values of this storage map.
+    pub(super) fn on_account_storage_set_map_item<S: ProcessState>(
+        &mut self,
+        process: &S,
+    ) -> Result<(), TransactionKernelError> {
+        // get slot index from the stack and make sure it is valid
+        let slot_index = process.get_stack_item(0);
+        if slot_index.as_int() as usize >= AccountStorage::NUM_STORAGE_SLOTS {
+            return Err(TransactionKernelError::InvalidStorageSlotIndex(slot_index.as_int()));
+        }
+
+        // get the KEY to which the slot is being updated
+        let new_map_key = [
+            process.get_stack_item(4),
+            process.get_stack_item(3),
+            process.get_stack_item(2),
+            process.get_stack_item(1),
+        ];
+
+        // get the VALUE to which the slot is being updated
+        let new_map_value = [
+            process.get_stack_item(8),
+            process.get_stack_item(7),
+            process.get_stack_item(6),
+            process.get_stack_item(5),
+        ];
+
+        let slot_index = slot_index.as_int() as u8;
+        if let Some(previous_maps_update) =
+            self.account_delta.storage.maps_updates.get_mut(&slot_index)
+        {
+            previous_maps_update.push((new_map_key, new_map_value))
+        } else {
+            self.account_delta
+                .storage
+                .maps_updates
+                .insert(slot_index, vec![(new_map_key, new_map_value)]);
+        }
```

Comment:
nit: idk if there's a posture regarding using `Entry`, but I think it it can be used here:

```suggestion
        self.account_delta.storage.maps_updates
            .entry(slot_index)
            .and_modify(|slot_map_updates| slot_map_updates.push((new_map_key, new_map_value)))
            .or_insert(vec![(new_map_key, new_map_value)]);
```

Reply:
ah nice, didn't know this yet